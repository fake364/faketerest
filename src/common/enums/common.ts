export enum GENDER_ENUM {
  FEMALE = -1,
  NOT_SPECIFIED = 0,
  MALE
}
