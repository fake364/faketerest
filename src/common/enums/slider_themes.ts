export enum SLIDER_THEMES {
	DINNER_IDEA = 'dinnerIdea',
	HOME_DECOR = 'homeDecor',
	NEW_OUTFIT = 'newOutfit',
	THUMB_IDEA = 'thumbIdea'
}
