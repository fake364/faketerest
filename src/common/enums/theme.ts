export enum THEME_TYPE {
	BASE = 'base',
	BUSINESS = 'business',
	NEWSROOM = 'newsroom'
}