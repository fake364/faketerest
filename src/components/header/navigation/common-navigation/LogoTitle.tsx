import React from 'React';
import Image from 'next/image';
import logoPic from '../../../../../public/logo.png';
import COMMON_CONSTANTS from '../../../../common/constants/commons';

type Props = {};

const LogoTitle: React.FC<Props> = () => {
	return (
		<div className="flex flex-col items-center justify-center min-h-screen py-2">
			<Image src={logoPic} alt={'Logo'} width={32} height={32} />
			<span className="underline bold">{COMMON_CONSTANTS.TITLE}</span>
		</div>
	);
};

export default LogoTitle;
