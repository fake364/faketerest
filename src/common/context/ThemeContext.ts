import React from 'react';
import { THEME_TYPE } from '../enums/theme';

const ThemeContext = React.createContext<{
	theme: THEME_TYPE;
	setTheme: Function;
}>({
	theme: THEME_TYPE.BASE,
	setTheme: () => null
});

export default ThemeContext;
