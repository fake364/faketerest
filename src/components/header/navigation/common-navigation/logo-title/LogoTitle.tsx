import React, { useContext } from 'react';
import Image from 'next/image';
import logoPic from '../../../../../../public/logo.png';
import COMMON_CONSTANTS from '../../../../../common/constants/commons';
import ThemeContext from '../../../../../common/context/ThemeContext';

type Props = { titleOff?: boolean; width?: number; height?: number };

const LogoTitle: React.FC<Props> = ({
  titleOff = false,
  height = 32,
  width = 32
}) => {
  const { theme } = useContext(ThemeContext);

  return (
    <div className="flex items-center">
      <Image src={logoPic} alt={'Logo'} width={width} height={height} />
      {!titleOff && (
        <span
          data-theme={theme}
          className="font-bold text-primary text-[20px] ml-[6px]"
        >
          {COMMON_CONSTANTS.TITLE}
        </span>
      )}
    </div>
  );
};

export default LogoTitle;
