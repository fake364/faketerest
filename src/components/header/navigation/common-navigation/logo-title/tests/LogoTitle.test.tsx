import React from 'react';
import { mount } from 'enzyme';
import ThemeContext from '../../../../../../common/context/ThemeContext';
import { THEME_TYPE } from '../../../../../../common/enums/theme';
import LogoTitle from '../LogoTitle';

jest.mock('../../../../../../../public/logo.png', () => () => '');
jest.mock('next/image', () => () => <div id="image" />);

describe('LogoTitle', () => {
  it('should setup theme for logo', () => {
    const wrapper = mount(
      <ThemeContext.Provider
        value={{ theme: THEME_TYPE.BASE, setTheme: jest.fn() }}
      >
        <LogoTitle />
      </ThemeContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
