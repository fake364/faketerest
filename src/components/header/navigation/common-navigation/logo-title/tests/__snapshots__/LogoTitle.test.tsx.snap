// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoTitle should setup theme for logo 1`] = `
<LogoTitle>
  <div
    className="flex items-center"
  >
    <Component
      alt="Logo"
      height={32}
      src={[Function]}
      width={32}
    >
      <div
        id="image"
      />
    </Component>
    <span
      className="font-bold text-primary text-[20px] ml-[6px]"
      data-theme="base"
    >
      Faketerest
    </span>
  </div>
</LogoTitle>
`;
